src/pages/settings/Profile/CustomStatus/StatusPage.js

```diff
@@ -0,0 +1,114 @@
+    const currentUserStatusText = lodashGet(currentUserPersonalDetails, 'status.text', '');
+    const draftEmojiCode = lodashGet(draftStatus, 'emojiCode');
+    const draftText = lodashGet(draftStatus, 'text');
+
+    const defaultEmoji = draftEmojiCode || currentUserEmojiCode;
+    const defaultText = draftText || currentUserStatusText;
+    const customStatus = defaultEmoji ? `${defaultEmoji} ${defaultText}` : '';
+    const hasDraftStatus = !!draftEmojiCode || !!draftText;
+
+    const updateStatus = useCallback(() => {
+        const endOfDay = moment().endOf('day').toDate();
+        User.updateCustomStatus({text: defaultText, emojiCode: defaultEmoji, clearAfter: endOfDay.toISOString()});
+
+        User.clearDraftCustomStatus();
+        Navigation.goBack(ROUTES.SETTINGS);
+    }, [defaultText, defaultEmoji]);
+
+    const clearStatus = () => {
+        User.clearCustomStatus();
+        User.clearDraftCustomStatus();
+    };
+
+    const footerComponent = useMemo(
+        () =>
+            hasDraftStatus ? (
+                <Button
+                    success
+                    text={localize.translate('statusPage.save')}
+                    onPress={updateStatus}
+                />
+            ) : null,
+        [hasDraftStatus, localize, updateStatus],
+    );
+
+    useEffect(() => () => User.clearDraftCustomStatus(), []);
+
+    return (
+        <StaticHeaderPageLayout
+            title={localize.translate('statusPage.status')}
+            onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS_PROFILE)}
+            backgroundColor={themeColors.midtone}
+            image={MobileBackgroundImage}
+            footer={footerComponent}
+        >
+            <View style={styles.m5}>
+                <Text style={[styles.textHeadline]}>{localize.translate('statusPage.setStatusTitle')}</Text>
+                <Text style={[styles.textNormal, styles.mt2]}>{localize.translate('statusPage.statusExplanation')}</Text>
+            </View>
+            <MenuItemWithTopDescription
+                title={customStatus}
+                description="Status"
+                shouldShowRightIcon
+                inputID="test"
+                onPress={() => Navigation.navigate(ROUTES.SETTINGS_STATUS_SET)}
+            />
+
+            {(!!currentUserEmojiCode || !!currentUserStatusText) && (
+                <MenuItem
+                    title={localize.translate('statusPage.clearStatus')}
+                    icon={Expensicons.Close}
+                    onPress={clearStatus}
+                    iconFill={themeColors.danger}
+                    wrapperStyle={[styles.cardMenuItem]}
+                />
+            )}
+        </StaticHeaderPageLayout>
+    );
+}
+
+StatusPage.displayName = 'StatusPage';
+StatusPage.propTypes = propTypes;
+
+export default compose(
+    withLocalize,
+    withCurrentUserPersonalDetails,
+    withOnyx({
+        status: {
```

Comment:
I think we should remove this and follow the pattern that we use for the profile page https://github.com/Expensify/App/blob/d416ddca7b7dcb0ced30ab2b1d5993c0a60b48da/src/pages/settings/Profile/ProfilePage.js#L62

Reply:
👍🏻